rfcs/0000-wifi-buddy-interface.md

```diff
@@ -0,0 +1,125 @@
+which are pinout compatible, so they can be used interchangeably. The ESP32-C3
+board will be treated as a system-on-module (SoM) and integrated with the MnemOS
+Beepy shield using pin headers (or potentially soldered onto the board).
+
+The following pins are broken out by the WiFi Buddy hardware and can be used for
+communication between the CPU and WiFi Buddy:
+
+- One SPI interface (`MOSI`, `MISO`, and `SCK`)
+- One I²C interface (`SDA` and `SCL`)
+- One UART (`TX` and `RX`)
+- Four additional GPIO pins
+
+### Software
+
+Communication between the CPU and the WiFi Buddy will include both a _control
+path_ and a _data path_. Data path communication refers to the actual network
+frames exchanged between the WiFi Buddy and the CPU, while control path
+communication consists of messages that configure the operation of the WiFi
+Buddy.
+
+Potential control path communication may include:
+
+- The CPU asking the WiFi Buddy to scan for WiFi access points
+- The CPU asking the WiFi Buddy to connect to or disconnect from a WiFi access
+  point
+- The ability for the CPU to reset the WiFi Buddy (?)
+- Information about the network interface, such as signal strength
+- Errors and other diagnostic information reported by the WiFi Buddy
+- `trace-proto`-formatted logs from the WiFi buddy (?)
+- Potentially, messages related to Bluetooth Low Energy operation (out of scope
+  for this document)
+
+It may also be desirable for the CPU to eventually be able to update the
+firmware on the WiFI buddy. However, for now, both of the target WiFi Buddy
+devices can be flashed over USB, so the MnemOS CPU being able to update the WiFi
+Buddy firmware is not a high priority at this time.
+
+Data path communication will be performed over the SPI bus, as it's the
+highest-bandwidth link available with this hardware. In [#121], we have
+tentatively concluded that the firmware will use a "dumb WiFi Buddy" design,
+where TCP or any other higher-level network protocols are implemented on the
+CPU, and the CPU and WiFi Buddy communicate by exchanging MAC-layer [802.3]
+Ethernet frames.
+
+## Design Questions
+
+Based on the background information discussed above, we can begin to enumerate
+design questions for the WiFi Buddy interface:
+
+- **Is control path communication in-band or out of band?** If data path frames
+  are exchanged between the host CPU and the WiFi Buddy over the SPI bus, do we
+  also exchange control messages on the same bus? Or, are control messages sent
+  using another interface, such as I²C?
+
+  Using a single interface for both control and data path communication is
+  conceptually simpler, and has the advantage of a clear and obvious ordering
+  between data and control messages. It also means that we don't necessarily
+  need to implement drivers for all the interfaces available on the ESP32-C3.
+
+  Downsides of in-band control messages include that we must introduce some
```

Comment:
Mention in chat: I think using postcard as the wire format means that decoding/encoding is likely to be low overhead (CPU and bandwidth use wise), and allow for flexibility in the future if we add more message types.